### PR TITLE
Build: Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by ygj6**
### Summary ###

Migrate CI to Github Action.

- Add `node.js.yaml` to support github action.

- Modify `Gruntfile.js` to support running travis and github actions at the same time.

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Use Node.js ${{ matrix.NODE_VERSION }}
+      uses: actions/setup-node@v2.1.2
+      with:
+        node-version: ${{ matrix.NODE_VERSION }}
+
+    - name: Install firefox esr
```
Comment (mgol) [inline .github/workflows/node.js.yml:54]:
```suggestion
    - name: Install firefox ESR
```

**Review by mgol — COMMENTED** [review]
Wow, this is so awesome, thanks a lot! We planned to migrate from Travis to GitHub Actions for quite some time.

I'll have a closer look this week.

The next step that we planned (to be done later) was to start testing on real browsers on GitHub Actions using our BrowserStack account, phasing out our [TestSwarm](http://swarm.jquery.org/project/jquery) usage. I was thinking about making it easy to restart runs only in some browsers if they're flaky. One advantage of TestSwarm is we're able to restart a test run of any module in any browser and leave the rest intact; it'd be great to have something like that here.

It'd also be great to have a pipeline graph view, where we're doing the full build only once and then split that to browser testing for each browser. Something like what [Blue Ocean](https://www.jenkins.io/projects/blueocean/) provides for Jenkins.

If you have any thoughts on the above that, I'm all ears.

In any case, thanks again for making a very important first step. :clap:

**Comment by ygj6** [comment]
Hi, mgol
Thanks you for your review.

I'm considering switch to Browerstack in other PR and keep this PR simply using Github Actions.
Is it Ok for you?

**Comment by mgol** [comment]
> I'm considering switch to Browerstack in other PR and keep this PR simply using Github Actions.
> Is it Ok for you?

Yes, definitely!

**Comment by mgol** [comment]
One thing that makes me worry a bit is missing support for re-running a single job. This is key to us; as we start testing on browsers, we need to be able to only re-run what failed instead of everything which creates a risk of yet another failure. See this thread: https://github.community/t/ability-to-rerun-just-a-single-job-in-a-workflow/17234

**Comment by mgol** [comment]
This feature missing makes me think we might want to try Circle CI with its visual workflows instead: https://circleci.com/docs/2.0/workflows/

**Comment by ygj6** [comment]
> One thing that makes me worry a bit is missing support for re-running a single job. This is key to us; as we start testing on browsers, we need to be able to only re-run what failed instead of everything which creates a risk of yet another failure. See this thread: https://github.community/t/ability-to-rerun-just-a-single-job-in-a-workflow/17234

It takes a short time to build jQuery. I think it is acceptable to re-run all jobs in github action.

**Comment by ygj6** [comment]
> I was thinking about making it easy to restart runs only in some browsers if they're flaky

As far as I known, we can get BrowserStack's report by karma. So we can parse the report to get the failed browsers and rerun the failed browsers in the karma service. Do you think I understand correctly? 


> Something like what Blue Ocean provides for Jenkins.

gh-action has planned [this feature ](https://github.com/github/roadmap/issues/88#issue-665304930)in Q4 milestones.

**Comment by mgol** [comment]
> As far as I known, we can get BrowserStack's report by karma. So we can parse the report to get the failed browsers and rerun the failed browsers in the karma service.

@ygj6 I have no idea how that would work and, if anything, it would be hacky. We need a native way to re-run a single job and we consider that to be a blocker for adopting any CI system.

**Review by mgol — CHANGES_REQUESTED** [review]
Marking as "Request changes" until our concerns with GitHub Actions are resolved.

**Review by Jdolf-Source — APPROVED** [review]

**Comment by mgol** [comment]
Closing & re-opening the PR to trigger the EasyCLA check...

**Review by mgol — APPROVED** [review]
Two things have happened since my last review:
1. GitHub Actions added support for re-running just the failed workflows.
2. We lost access to Travis 😢 
so I decided to look again at this PR, rebased it & tweaked very slightly and it looks like it works!

Thanks a lot for the PR!

**Comment by mgol** [comment]
Landed on `main` in e23190e63cb121da79b92e6641a81a44dcea9252 and on `3.x-stable` in b39cfa1505f5ccce28a4a83d8700f0edb3e9cd8c.

**Pull request merged**